# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLMediaElement.cpp
# Commit: e2dc8fd47bc2
# Full Hash: e2dc8fd47bc2873b9bd7b17ccc643de77a82d884
# Author: alwu <alwu@mozilla.com>
# Date: 2020-07-24 09:32:06
# Regressor Bug: 1653390
# File Overlap Count: 1
# Description:
#   Bug 1653390 - part1 : remove start check. r=bryce
#   
#   The reason of checking those conditions before calling `StartListeningMediaControlKeyIfNeeded()` is because we was only starting the listener after media starts, which has been modified in D79234.  Now we would start the listener before media starts, so we can remove that check.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D84114
# ==============================================================================

diff -r cce1c9388eeb -r e2dc8fd47bc2 dom/html/HTMLMediaElement.cpp
--- a/dom/html/HTMLMediaElement.cpp	Fri Jul 24 05:48:31 2020 +0300
+++ b/dom/html/HTMLMediaElement.cpp	Thu Jul 23 21:32:43 2020 +0000
@@ -6546,11 +6546,7 @@
         !AutoplayPolicy::IsAllowedToPlay(*this)) {
       MaybeNotifyAutoplayBlocked();
     }
-    // If we stopped listening to the event when we suspended media element,
-    // then we should restart to listen to the event if we haven't done so yet.
-    if (mMediaControlKeyListener && !mMediaControlKeyListener->IsStarted()) {
-      StartListeningMediaControlKeyIfNeeded();
-    }
+    StartListeningMediaControlKeyIfNeeded();
   }
   if (StaticPrefs::media_testing_only_events()) {
     auto dispatcher = MakeRefPtr<AsyncEventDispatcher>(
